ql/errors.hpp

```diff
@@ -49,94 +46,58 @@ namespace QuantLib {
         const char* what() const noexcept override;
 
       private:
-        ext::shared_ptr<std::string> message_;
+        std::string message_;
```

Comment:
Technically, having `ext::shared_ptr<std::string>` makes the copy constructor not throwing, as required by the standard.  With `std::string`, it can throw.  On the other hand, if it throws, it probably means we're out of memory anyway so all bets are off.  I'd try and measure the effect of `shared_ptr` only, though.

Reply:
Right, I had to look that requirement to be honest.